.github/workflows/test.yaml

```diff
@@ -52,27 +52,89 @@ jobs:
-        with:
-          version: 9
-      - name: Change directory to examples/cloudflare-workers
-        run: cd examples/cloudflare-workers
-
       - name: Install Dependencies
-        run: pnpm install
+        run: bun install
         working-directory: examples/cloudflare-workers
 
       - name: Install local package
-        run: pnpm install
+        run: bun install @upstash/qstash@file:../../dist
         working-directory: examples/cloudflare-workers
 
-      - name: Local build
-        run: sh ./test.sh
+      - name: Change main file to ci.ts
+        run: sed -i 's/main = "src\/index.ts"/main = "src\/ci.ts"/' wrangler.toml
+        working-directory: examples/cloudflare-workers
+
+      - name: Add environment
+        run: |
+          echo '[vars]' >> wrangler.toml
+          echo "QSTASH_TOKEN = \"$QSTASH_TOKEN\"" >> ./wrangler.toml
         working-directory: examples/cloudflare-workers
 
+      - name: Run example
+        run: bun dev &
+        working-directory: examples/cloudflare-workers
+
+      - name: Test
+        run: bun test ci.test.ts
+        working-directory: examples/cloudflare-workers
+        env:
+          DEPLOYMENT_URL: http://127.0.0.1:8787
+
+  cloudflare-workers-deployed:
+    concurrency: cloudflare-workers-deployed
+    needs:
+      - release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup repo
+        uses: actions/checkout@v3
+      - name: Setup nodejs
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Setup Bun
+        uses: oven-sh/setup-bun@v1
+        with:
+          bun-version: latest
+
+      - name: Install example
+        run: |
+          bun add @upstash/qstash@${{needs.release.outputs.version}}
+          npm i -g wrangler
+        working-directory: examples/cloudflare-workers
+
+      - name: Add account ID
+        run: echo 'account_id = "${{ secrets.CLOUDFLARE_ACCOUNT_ID }}"' >> wrangler.toml
+        working-directory: examples/cloudflare-workers
+
+      - name: Add environment
+        run: |
+          echo '[vars]' >> wrangler.toml
+          echo "QSTASH_TOKEN = \"$QSTASH_TOKEN\"" >> ./wrangler.toml
+        working-directory: examples/cloudflare-workers
+
+      - name: Change main file to ci.ts
+        run: sed -i 's/main = "src\/index.ts"/main = "src\/ci.ts"/' wrangler.toml
+        working-directory: examples/cloudflare-workers
+
+      - name: Deploy
+        run: wrangler publish
+        working-directory: examples/cloudflare-workers
+        env:
+          CLOUDFLARE_API_TOKEN: ${{secrets.CLOUDFLARE_API_TOKEN}}
+
+      - name: Test
+        run: bun test ci.test.ts
+        working-directory: examples/cloudflare-workers
+        env:
+          DEPLOYMENT_URL: https://upstash-qstash.upsdev.workers.dev
+
   nextjs-local-build:
     runs-on: ubuntu-latest
+    # disabled because of a local build issue. Will be tested properly in
+    # the deployment test
+    if: false
```

Comment:
What is the local build issue?

Reply:
similar to https://github.com/upstash/ratelimit-js/pull/123#discussion_r1789718986

it works fine in release, but local installation has a type error. I don't want to write @ts-expect-error in order to not miss it in release, so I disabled the local one until we figure this out